src/components/Drawer/components/DrawerItem.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+
+import {useForkRef} from '../../../hooks';
+import {block} from '../../utils/cn';
+import {useResizableDrawerItem} from '../hooks/useResizableDrawerItem';
+import type {DrawerDirection, OnResizeHandler} from '../hooks/useResizeHandlers';
+
+const b = block('drawer');
+
+import './Drawer.scss';
+
+export interface DrawerItemProps {
+    /** Content to be displayed within the drawer item. */
+    children?: React.ReactNode;
+
+    /** Determines whether the drawer item is visible or hidden. */
+    visible?: boolean;
+
+    /**
+     * Specifies the direction from which the drawer should slide in, `left` by default.
+     * @default left
+     */
+    direction?: DrawerDirection;
+
+    /** Additional custom class name applied to the drawer item. */
+    className?: string;
+
+    /** Determines whether the drawer item can be resized */
+    resizable?: boolean;
+
+    /**
+     * The width of the resizable drawer item.
+     * If not provided, the width will be stored internally.
+     */
+    width?: number;
+
+    /** Called at the start of resizing. */
+    onResizeStart?: OnResizeHandler;
+
+    /** Called at the end of resizing. */
+    onResizeEnd?: OnResizeHandler;
+
+    /**
+     * Called at the end of resizing. Can be used to save the new width.
+     * @param width The new width of the drawer item
+     */
+    onResize?: OnResizeHandler;
+
+    /** The minimum width of the resizable drawer item */
+    minResizeWidth?: number;
+
+    /** The maximum width of the resizable drawer item */
+    maxResizeWidth?: number;
+
+    disablePortal?: boolean;
+
+    /** Optional inline styles to be applied to the DrawerItem component. */
+    style?: React.CSSProperties;
+}
+
+export const DrawerItem = React.forwardRef<HTMLDivElement, DrawerItemProps>(
+    function DrawerItem(props, ref) {
+        const {
+            visible,
+            children,
+            direction = 'left',
+            className,
+            resizable,
+            width,
+            minResizeWidth,
+            maxResizeWidth,
+            onResizeStart,
+            onResizeEnd,
+            onResize,
+            style = {},
+            disablePortal,
+            ...rest
+        } = props;
+
+        const [isInitialRender, setInitialRender] = React.useState(true);
```

Comment:
State doesn't update anywhere

Reply:
it is being updated inside a useEffect hook, 20 lines below